### ISSUE TITLE
`Menu`s are not keyboard-accessible

### ISSUE THREAD

**Issue by sdnts** [prisma/lens#127]
You should be able to:

- `Tab` your way to the menu trigger element
- Press `Down` / `Space` / `Enter` to open the menu
- Once opened, you should be able to move around the selection using arrow keys
- Once opened, you should be able to press `Esc` to close the menu and return focus to the trigger. No `Menu.Option` `onAction`s should be fired.
- Once opened, upi should be able to press `Space` / `Enter` while on a `Menu.Option` and fire that option's `onAction` callback. The menu should also close.

Some of these work, others are intermittent / do not work at all.

**Comment by nilubava** [comment]
@luanvdw Could you pls review this and close the issue if stale
